src/states/elder/mod.rs

```diff
@@ -404,83 +421,147 @@ impl Elder {
+                    }
+                    AccumulatingEvent::AckMessage(ref payload) => {
+                        our_pfx.matches(&payload.dst_name)
+                            && !completed_events.contains(&event.payload)
+                    }
 
-                // Drop: no longer relevant after prefix change.
-                // TODO: verify this is really the case. Some/all of these might still make sense
-                // to carry over. In case it does not, add a comment explaining why.
-                AccumulatingEvent::StartDkg(_)
-                | AccumulatingEvent::ParsecPrune
-                | AccumulatingEvent::Relocate(_) => false,
+                    // Drop: no longer relevant after prefix change.
+                    // TODO: verify this is really the case. Some/all of these might still make sense
+                    // to carry over. In case it does not, add a comment explaining why.
+                    AccumulatingEvent::StartDkg(_)
+                    | AccumulatingEvent::ParsecPrune
+                    | AccumulatingEvent::Relocate(_) => false,
+
+                    // Keep: Additional signatures for neighbours for sec-msg-relay.
+                    AccumulatingEvent::SectionInfo(ref elders_info, _)
+                    | AccumulatingEvent::NeighbourInfo(ref elders_info) => {
+                        our_pfx.is_neighbour(elders_info.prefix())
+                    }
 
-                // Keep: Additional signatures for neighbours for sec-msg-relay.
-                AccumulatingEvent::SectionInfo(ref elders_info, _)
-                | AccumulatingEvent::NeighbourInfo(ref elders_info) => {
-                    our_pfx.is_neighbour(elders_info.prefix())
+                    // Keep: Still relevant after prefix change.
+                    AccumulatingEvent::TheirKeyInfo(_)
+                    | AccumulatingEvent::SendAckMessage(_)
+                    | AccumulatingEvent::User(_) => true,
                 }
-
-                // Keep: Still relevant after prefix change.
-                AccumulatingEvent::TheirKeyInfo(_)
+            })
+            .partition(|event| match event.payload {
+                // Events to re-insert
+                AccumulatingEvent::Offline(_)
+                | AccumulatingEvent::AckMessage(_)
+                | AccumulatingEvent::StartDkg(_)
+                | AccumulatingEvent::ParsecPrune
+                | AccumulatingEvent::Relocate(_)
+                | AccumulatingEvent::SectionInfo(_, _)
+                | AccumulatingEvent::NeighbourInfo(_)
+                | AccumulatingEvent::TheirKeyInfo(_)
                 | AccumulatingEvent::SendAckMessage(_)
                 | AccumulatingEvent::User(_) => true,
-            })
-            .for_each(|event| {
-                self.vote_for_network_event(event.clone());
+                // Events to re-process
+                AccumulatingEvent::Online(_) => false,
```

Comment:
I don't think this is guaranteed to be consensused, because the `cached_events` collection is extended from unpolled observations too.

Reply:
Sorry, I'm not clear what your mean? what is the impact?

Reply:
Let me go through the rest of the commit. Maybe it is OK, not sure yet.

Reply:
OK, the problem I think is that this `Online` event might have come from `our_unpolled_observations` which means it's not necessarily reached parsec consensus. But we are passing it to the `to_process` collection which is later passed to `process_post_reset_events` where it is actioned. 

On the other hand, we only send `BootstrapResponse` there which doesn't require consensus anyway, so perhaps this is OK? Might be worth adding a short explanation comments, because this can confuse people.

Reply:
Ok, I'll add a small comment to the struct

Reply:
/// The cached events that should be revoted. Not shared state as only contains the one we voted for.
    /// Also contains the one we voted for but never reached consensus.